src/OrchardCore.Modules/OrchardCore.Lucene/LuceneIndexingService.cs

```diff
@@ -50,21 +53,28 @@ public class LuceneIndexingService
 
         public ILogger Logger { get; }
 
-        public async Task ProcessContentItemsAsync()
+        public async Task ProcessContentItemsAsync(string indexName = default)
         {
             // TODO: Lock over the filesystem in case two instances get a command to rebuild the index concurrently.
-
             var allIndices = new Dictionary<string, int>();
+            var lastTaskId = Int32.MaxValue;
 
-            // Find the lowest task id to process
-            var lastTaskId = int.MaxValue;
-            foreach (var indexName in _indexManager.List())
```

Comment:
Here it was returning all indices by looking at folders located in the /App_Data/Lucene/ folder and return a list of folders name. So every time ProcessContentItemsAsync was called it was indexing all indices.